homeassistant/components/modbus/binary_sensor.py

```diff
@@ -99,18 +102,33 @@ def device_class(self) -> Optional[str]:
         """Return the device class of the sensor."""
         return self._device_class
 
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
     def update(self):
         """Update the state of the sensor."""
-        if self._input_type == INPUT_TYPE_COIL:
-            result = self._hub.read_coils(self._slave, self._address, 1)
-        else:
-            result = self._hub.read_discrete_inputs(self._slave, self._address, 1)
+
+        def _set_unavailable():
+            if self._available:
```

Comment:
Make this a guard clause. `if not self._available: return`

Reply:
Done.